core/app/[locale]/(default)/product/[slug]/static/page.tsx

```diff
@@ -56,5 +57,42 @@ export async function generateStaticParams() {
   });
 }
 
+interface Props {
+  params: Promise<{ slug: string; locale: LocaleType }>;
+}
+
+export async function generateMetadata(props: Props): Promise<Metadata> {
```

Comment:
This one took a while to figure out, turns our our `generateMetadata` from the non-static page which we used to re-export is using `searchParams` which marks the page as dynamic even if we use `force-static`.

Hard to track as this doesn't fail on dev, only in production builds.

Had to duplicate this fn and remove the usage of searchParams, since it's a static page we never hit it if we have searchParams set anyway.

I would like to know how this used to work in Next 14 and if there are plans to add `searchParams` to the list of `force-static` and return empty on static builds.